### ISSUE TITLE
methylkit unite() function reassigns sample ID's

### ISSUE THREAD

**Issue by laurahspencer** [RobertsLab/resources#1648]
I'm encountering an issue with methylkit where it re-assigns sample IDs!  I'm running methylkit's unite() function, which takes a list of dataframes (in a "methylRaw" object) each containing methylation data (coverage, numCs, numTs) for a sample, and column binds them into one dataframe (in a "methylBase" object) while also filtering them to retain loci with coverage across all samples (or a min # of samples set by the user).  The new dataframe/methylBase object should contain 3 columns of data for each sample with the following naming convention: 

```
coverage5   numCs5    numTs5   
```
where the numbers (here 5) represent the sample ID, which is specified in the methylRaw object that you feed into the unite() function.  The issue is that sample IDs specified in the methylRaw object are not correctly being assigned by unite().  I have 13 samples, with the IDs 1, 2, 3, 4, 7, 8, 10, 11, 12, 15, 17, 18, 20 (i.e. not all consecutive numbers). When I run unite() the resulting columns of data contain sample IDs #1-13.  Thinking that unite() renamed them consecutively I tried to spot-check the data to see if I can confidently rename the columns using my original IDs, but the data doesn't line up (possibly because I've destranded, but I can't be sure).  Has anyone run into this issue?  

Here is [zip file](https://drive.google.com/file/d/1wL-57830DtPdmmei2z0qZknGVqpOOupY/view?usp=sharing) containing the methylRaw object `meth_10x`, and using the code `methylKit::unite(meth_10x, destrand=TRUE)` I created the methylBase object `meth_unite_10x`.

**Comment by kubu4** [comment]
Do you have link to repo with full code that we can follow along with?

**Comment by kubu4** [comment]
Also, any chance you could put the file on Gannet (or Owl)? The links to files hosted on Google Drive aren't really useful when trying to work on a remote computer, as those links don't work with `wget` or `curl` command line programs.

**Comment by laurahspencer** [comment]
Here is my RMarkdown file, [01_methylation.Rmd](https://github.com/laurahspencer/DuMOAR/blob/main/scripts/01_methylation.Rmd#L640) where I run the data through methylkit.  Line 640 is where I run the unite() function on the methylRaw object "meth_10x" to create the methylBase object "meth_unite_10x".  

The two R objects are now also located in the project repo, [results/methylkit/unite_issue](https://github.com/laurahspencer/DuMOAR/tree/main/results/methylkit/unite_issue), does that work?

**Comment by laurahspencer** [comment]
I started an issue on the methylkit github repo https://github.com/al2na/methylKit/issues/283 so we'll see what they say

**Comment by laurahspencer** [comment]
Heard from the developers. The dataframe created by the unite() function does NOT use the sample IDs to name the columns containing data. Instead ... 

"These are actually the column names of the internal dataframe. They are not
generated based on the given sample ids, nor the treatment vector. They are
simply referring to the n-th sample in the object, or the n-th position of
the sample-id vector."

So, columns "coverage1", "numCs1", and "numTs1" don't necessarily contain data from sample "1", but from the 1st sample listed in the vector of sample IDs.  To see the vector of sample IDs, one can use `getSampleID()` on either the methylRaw object or methylBase object.  If one uses `getData()` to extract the methylBase object created by `unite()` as a simple dataframe it won't have the correct sample IDs. This shouldn't create a problem if all sample IDs are numbered sequentially, but I ran into an issue when I removed a few samples.  

NOTE: If you use the `percMethylation()` function on the methylBase object and convert that to a dataframe it will have the correct sample IDs.
